File: frontend/src/types/application/ApplicationDetailsDto.ts
import { ApplicationSearchDto } from '@/types/application/ApplicationSearchDto.ts'
import { ApplicationTransitionDto } from '@/types/application/ApplicationTransitionDto.ts'
import { ApplicationHistoryDto } from '@/types/application/ApplicationHistoryDto.ts'

export type ApplicationDetailsDto = ApplicationSearchDto & {
  history: ApplicationHistoryDto[]
  transitions: ApplicationTransitionDto[]
}

File: backend/src/main/kotlin/ch/streckeisen/mycv/backend/application/dto/ApplicationDetailsDto.kt
package ch.streckeisen.mycv.backend.application.dto

import com.fasterxml.jackson.annotation.JsonProperty
import java.time.LocalDateTime

data class ApplicationDetailsDto(
    val id: Long,
    val jobTitle: String,
    val company: String,
    val status: ApplicationStatusDto,
    val source: String?,
    val description: String?,
    val createdAt: LocalDateTime,
    val updatedAt: LocalDateTime?,
    val history: List<ApplicationHistoryDto>,
    val transitions: List<ApplicationTransitionDto>,
    @field:JsonProperty(value = "isArchived")
    val isArchived: Boolean
)